app/controllers/colleges_controller.rb

```diff
@@ -0,0 +1,58 @@
+class CollegesController < ApplicationController
+
+  def index
+    @college_list = College.all
+  end
+
+  def new
+    @college = College.new
+  end
+
+  def show
+    @college = College.find_by_id(params[:id]) or not_found
+  end
+
+  def create
+    @college = College.new(college_param)
+    if @college.save
+      redirect_to colleges_path
+    else
+      render 'new'
```

Comment:
In case of errors, use _rails flash notifier_ instead of rendering the same page